pkg/collectors/daemonset.go

```diff
@@ -99,11 +100,21 @@ func (l DaemonSetLister) List() ([]v1beta1.DaemonSet, error) {
 	return l()
 }
 
-func RegisterDaemonSetCollector(registry prometheus.Registerer, informerFactories []informers.SharedInformerFactory, opts *options.Options) {
+func RegisterDaemonSetCollector(registry prometheus.Registerer, informerFactories map[string][]interface{}, opts *options.Options) {
 
 	infs := SharedInformerList{}
-	for _, f := range informerFactories {
-		infs = append(infs, f.Extensions().V1beta1().DaemonSets().Informer().(cache.SharedInformer))
+	gvr := schema.GroupVersionResource{
+		Group:    "extensions",
```

Comment:
Passing these as a struct of strings makes it easier to generalize in the long run, but does not catch typos during compile time. Not a blocker, just something to keep in mind.

Reply:
👌